botocove/cove_runner.py

```diff
@@ -78,6 +79,12 @@ def cove_thread(
             return cove_session.format_cove_result(result)
 
         except Exception as e:
+
+            if isinstance(e, EndpointConnectionError):
+                if not cove_session.is_region_in_boto3_model():
+                    logger.exception(cove_session.format_cove_error(e))
+                    raise InvalidRegion(account_session_info["Region"]) from e
```

Comment:
It may not be immediately obvious that an EndpointConnectionError is caused by an invalid region. You can read the region name from the error message, but it's a bit harder to parse. If you think we don't need the custom exception I can remove it.